queries.sql

```diff
@@ -9,6 +9,25 @@ SELECT * FROM animals WHERE neutered=TRUE;
 SELECT * FROM animals WHERE name!='Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+/* Inside a transaction update the animals table by setting the species column to unspecified. Then roll back  */
+BEGIN;
+UPDATE animals
+SET species = 'Unspecified';
+ROLLBACK;
+
+/* Name Species */
+BEGIN;
+UPDATE animals
+SET species='digimon' 
+WHERE
+name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE name NOT LIKE '%mon';
+
+COMMIT WORK;
```

Comment:
Hello,

It's great to see that you've implemented the transaction to update the `animals` table according to the specified conditions. However, there are some aspects of your code that need adjustment to fully meet the project requirements:

1. **Second UPDATE Condition:**
   According to the project requirements, the second `UPDATE` should set the species column to 'pokemon' for all animals that don't already have a species set. Your current code sets the species to 'pokemon' for all names not ending in 'mon', without checking if the species is already set. We need to correct this to only update rows where the `species` column is null or unspecified.

2. **Verification Step Missing:**
   The requirements explicitly ask for verification that the changes were made before and after the transactions are committed. This would typically involve a `SELECT` statement after the `UPDATE` statements before and after the `COMMIT` to ensure that the updates are correct.

3. **Commit Syntax [OPTIONAL]:**
   The `COMMIT WORK;` statement is valid, but `WORK` is optional in PostgreSQL. You can simply use `COMMIT;`.

Here is an improved version of your code with these considerations:

```sql
   BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

Please make sure to include the verification queries after your update statements and correct the condition for the second update to reflect the requirement accurately. After running your transaction with the verification step, if the output is as expected, then you can proceed to commit the transaction.

Keep up with the good work, and remember that verifying your updates before committing is a crucial step in transaction management!